03week/towersOfHanoi.js

```diff
@@ -13,33 +13,60 @@ let stacks = {
   c: []
 };
 
-function printStacks() {
+const printStacks = () => {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+const movePiece = (startStack, endStack) => {
+  //use push and pop to move the pieces from stack to stack
+  return stacks[endStack.toString()].push(stacks[startStack.toString()].pop());
 }
 
-function isLegal() {
-  // Your code here
-
-}
+//a legal move in towersOfHanoi can only be made if there is an empty stack or if
+//the destPiece is larger than the startPiece
+const isLegal = (startStack, endStack) => {
+  // Use length to find the end element in each array to compare against and check if legal move
+  const startPiece = stacks[startStack.toString()][stacks[startStack.toString()].length-1];
+  const destPiece = stacks[endStack.toString()][stacks[endStack.toString()].length-1];
 
-function checkForWin() {
-  // Your code here
+  //The startPiece can move to an empty stack or on top of a destPiece that is larger
+  return stacks[endStack.toString()].length === 0 || startPiece < destPiece;
 
 }
 
-function towersOfHanoi(startStack, endStack) {
-  // Your code here
+//Win state for towersOfHanoi means stack b = [4,3,2,1]
+const checkForWin = () => {
+  //We will join stack b and check if it's equal to the winning joined string
+  return stacks['b'].join('') === '4321';
```

Comment:
what about stack c? an easier way would be to just check if the length of b or c is 4